azuredevops/pull-request-gate.yml

```diff
@@ -0,0 +1,42 @@
+trigger: none # No CI build
+
+schedules:
+- cron: "0 4 * * *" # Time is UTC
+  displayName: Daily build
+  branches:
+    include:
+      - master
+      - dev
+  always: true
+
+pool:
+  vmImage: 'ubuntu-latest'
+
+jobs:
+- job: Linux
+  pool:
+    vmImage: 'ubuntu-16.04'
+  strategy:
+    maxParallel: 2
+    matrix: # Would like to grab this from the Library variables
```

Comment:
I would like to be able to generate this matrix from a variable. The variable would be set to something like "3.5,3.6"

Reply:
I see what you're doing.

A few things:
- I usually end up splitting windows and Linux into their own pipelines for maintenance convenience. But, I suggest to keep them in the same pipeline for now until one or the other fails at a different rate.

- I believe if you do something like this:
jobs:
- template: jobs/build.yml  #Template ref
  parameters:
     name: Linux
    Pool: ubuntu-xxxx

then in the template, use a matrix for python versions, that that may simplify.
since the template can be used for both Linux and windows, that will make it cleaner.
If there are differences needed btw windows and Linux in the template, I can show you an example.

then in the template, set up it's own parameters and jobs
     vmImage: 'ubuntu-16.04'

Reply:
Not quite set in a variable, but the latest iteration takes the above idea of a job template and is able to generate its matrix